transcom/mymove
# B 18031 prime UI update sit service items

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A870433&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

In order to simulate the prime using the `PATCH` endpoint `updateMTOServiceItem`, we needed to allow for this functionality in the prime simulator to allow for ease of testing. At this time this functionality is only allowed for SIT service items that have the service codes:
- `DOFSIT`
- `DOASIT`
- `DOPSIT`
- `DDDSIT`

The values that can be updated:
- `sitDepartureDate`
- `sitRequestedDelivery`
- `sitCustomerContacted`
- `sitDestinationFinalAddress` (destination SIT items only)


The following was completed in this PR:
- improved UI at first glance when viewing service items to display service code + service item name
- added new path to handle displaying new forms/components
- created components for destination and origin SIT service items, as well as a parent component that holds logic on displaying the respective components depending on the service item's `reServiceCode`
- updated `swagger-def` & `model_to_payload.go` for origin SIT service items to send the values that were requested to be updated so they can populate in the UI
- created test files for newly created components, updated existing tests that were impacted with new changes
- added `updateMTOServiceItem` to `primeApi.js` so it can be used in the components (was not previously implemented)
- updated button that requested update for final address from `Update Service Items` to `Request Destination SIT Address Change` to improve clarify of button function

*NOTE*
When updating the final destination address for a destination SIT service item, if an existing address is populated in the form then it will NOT be updated if new values are put in. If the service item does not have a final destination address yet, then it will update accordingly. 
This is backend work and is out of scope of this PR - will be addressed at a later date.


### How to test

1. Fire up `officelocal:3000`
2. Login as a the prime simulator
3. Click on a move with service items
4. Scroll down and you'll see an "Edit" button next to SIT service items that can be edited
5. Depending on if it's origin or destination SIT, the respective component will render
6. Fill out the form with the values and submit
7. The service item should be updated in the db, pending no errors

## Screenshots
![Screenshot 2023-12-11 at 10 24 06 AM](https://github.com/transcom/mymove/assets/136510600/befa3382-b500-4d08-964d-b0308d41a6cd)

![Screenshot 2023-12-12 at 11 20 58 AM](https://github.com/transcom/mymove/assets/136510600/16799ea6-ff6f-4546-9e98-e829082a0c27)

![Screenshot 2023-12-11 at 10 49 34 AM](https://github.com/transcom/mymove/assets/136510600/375852e0-43bd-4f98-af3f-1003c34d34d4)
